### PR TITLE
triggers: Let mcontrol.match be default (0/equal) if maskmax is 0

### PR DESCRIPTION
**Pull request by YenHaoChen**
This PR fixes the issue in https://github.com/riscv-software-src/riscv-isa-sim/issues/1785.

Specifically, the spec requires unsupporting ```mcontrol.match=NAPOT``` if ```mcontrol.maskmax=0```.
image

### CODE REVIEW

File: riscv/triggers.cc [hunk 1]
```diff
@@ -294,13 +294,14 @@ void mcontrol6_t::tdata1_write(processor_t * const proc, const reg_t val, const
   auto xlen = proc->get_const_xlen();
   assert(get_field(val, CSR_MCONTROL6_TYPE(xlen)) == CSR_TDATA1_TYPE_MCONTROL6);
   dmode = get_field(val, CSR_MCONTROL6_DMODE(xlen));
+  const reg_t maskmax6 = xlen - 1;
```
Comment (YenHaoChen) [inline riscv/triggers.cc:297]: @rtwfroody I didn't find an explicit upper bound for maskmax6 in the Debug spec. Does the spec intend ```maskmax6=31``` in RV32 and ```maskmax6=63``` in RV64 for maximum?

File: riscv/triggers.cc [hunk 2]
```diff
@@ -244,11 +244,11 @@ std::optional<match_result_t> mcontrol_common_t::detect_memory_access_match(proc
   return std::nullopt;
 }
 
-mcontrol_common_t::match_t mcontrol_common_t::legalize_match(reg_t val) noexcept
+mcontrol_common_t::match_t mcontrol_common_t::legalize_match(reg_t val, reg_t maskmax) noexcept
 {
   switch (val) {
+    case MATCH_NAPOT: if (maskmax == 0) return MATCH_EQUAL;
     case MATCH_EQUAL:
-    case MATCH_NAPOT:
```
Comment (YenHaoChen) [inline riscv/triggers.cc:251]: This seems tricky. Invite comment on this modification.

File: riscv/triggers.cc (hunk 1 above)
Comment (rtwfroody) [reply riscv/triggers.cc:297]: Yes. It doesn't make sense for maskmax6 to be any larger than that, because then you're just doing an equal comparison.

**Review by rtwfroody — APPROVED** [review]
This looks OK to me.

To be clear, it explicitly removes napot support from mcontrol, and makes it work for mcontrol6.

**Review by aswaterman — APPROVED** [review]
Code looks OK; I didn't review the functionality in detail.

**Pull request merged**